repository/src/main/java/org/apache/atlas/tasks/TaskManagement.java

```diff
@@ -251,6 +261,16 @@ private void startInternal() {
             return;
         }
 
+        String currentHostName = System.getenv(HOST_NAME);
+        String taskExecutionPodName = AtlasConfiguration.TASK_EXECUTION_POD_HOST_NAME.getString();
+
+        if (!taskExecutionPodName.isEmpty() && currentHostName != null) {
+            if (currentHostName.equals(taskExecutionPodName))
+                setTaskExecutionInProgressInSeparation(true);
```

Comment:
should we set serviceState.getState() to EXECUTING_TASK here?

Reply:
I would like to do that but ServiceState class is not accessible from atlas-repository. When I imported that causing cyclic module. As it was already  dependency of atlas-webapp

Reply:
Can we do that in nay of the filter then? I will check & get back on this shortly